### PR TITLE
CDAP-2331: Respond with 404 when listing apps in non-existent namespace

### PR DESCRIPTION
**Pull request by jwang47**
https://issues.cask.co/browse/CDAP-2331
http://builds.cask.co/browse/CDAP-DUT2470

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java [hunk 1]
```diff
@@ -171,22 +172,19 @@ protected int getInstances(HttpRequest request) throws IllegalArgumentException,
...
+    }
 
-      responder.sendJson(HttpResponseStatus.OK, appRecords);
-    } catch (SecurityException e) {
-      LOG.debug("Security Exception while retrieving app details: ", e);
-      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
-    } catch (Throwable e) {
-      LOG.error("Got exception : ", e);
-      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+    List<ApplicationRecord> appRecords = Lists.newArrayList();
```
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java]: better to use `new ArrayList<>()` Java 7 style?

**Comment by sagarkapare** [comment]
@jwang47 code looks good. Is there any unit test for this already?

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java (hunk 1 above)
Comment (jwang47) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java]: Oops. Yes.

**Comment by jwang47** [comment]
Will add a unit test.

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java [hunk 2]
```diff
@@ -171,22 +172,19 @@ protected int getInstances(HttpRequest request) throws IllegalArgumentException,
...
-      }
+  protected final void getAppRecords(HttpResponder responder, Store store,
+                                     String namespaceId) throws NamespaceNotFoundException {
+    Id.Namespace namespace = Id.Namespace.from(namespaceId);
+    if (store.getNamespace(namespace) == null) {
+      throw new NamespaceNotFoundException(namespace);
+    }
 
-      responder.sendJson(HttpResponseStatus.OK, appRecords);
-    } catch (SecurityException e) {
```
Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java]: these exceptions are no longer applicable here?

Comment (jwang47) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java]: Yeah, now HttpExceptionhandler handles these exceptions.

**Pull request merged**